projects/core/src/events/event.service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable, Type } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { CxEvent } from './event.model';
+import { EventRegister } from './event.register';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EventService {
+  constructor(protected eventRegiser: EventRegister) {}
+
+  on<T>(eventClass: Type<T>): Observable<CxEvent<T>> {
```

Comment:
Not sure why not 
```
  on<T>(eventClass: Type<T>): Observable<T> {
    return this.eventRegiser
      .getValue(eventClass);
  }
```
as T is an actual class, that extends from CxEvent, it seems we are unnecessary wrapping it with in another cxEvent.